lib/jekyll/jekyll-sitemap.rb

```diff
@@ -8,20 +8,20 @@ class JekyllSitemap < Jekyll::Generator
     # Main plugin action, called by Jekyll-core
     def generate(site)
       @site = site
-      unless sitemap_exists?
-        write
-        @site.keep_files ||= []
-        @site.keep_files << "sitemap.xml"
-      end
+      @site.pages << sitemap_content unless sitemap_exists?
     end
 
+    private
```

Comment:
Technically, technically, this would be a breaking change to the Ruby API. I'm all for it, but we'd have to make it a major bump then.